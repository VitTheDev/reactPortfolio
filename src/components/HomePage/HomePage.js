import React from 'react';
import { Link } from 'react-router-dom';

import './HomePage.scss';

const HomePage = () => {
  return (
    <div className="HomePage">
      <h2>Hi, I am Vit and I specialise in Frontend Web Development.</h2>

      <Link to="/mywork">
        <button className="button">See My Work</button>
      </Link>
    </div>
  );
};

export default HomePage;
