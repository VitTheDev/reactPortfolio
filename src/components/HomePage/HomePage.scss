.HomePage {
  width: 100%;
  height: 100%;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  margin: 2rem auto;
  color: #fff;
  font-size: 1.5rem;
  animation: fadeIn 1s linear;
}

@keyframes fadeIn {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.button {
  margin: 4rem 2rem;
  min-width: 300px;
  min-height: 60px;
  font-size: 22px;
  text-transform: uppercase;
  letter-spacing: 1.3px;
  font-weight: 700;
  color: #313133;
  background: #4fd1c5;
  background: linear-gradient(
    90deg,
    rgba(129, 230, 217, 1) 0%,
    rgba(79, 209, 197, 1) 100%
  );
  border: none;
  border-radius: 1000px;
  box-shadow: 12px 12px 24px rgba(79, 209, 197, 0.64);
  transition: all 0.3s ease-in-out 0s;
  outline: none;
  position: relative;
  padding: 10px;
}

.button::before {
  content: '';
  border-radius: 1000px;
  min-width: calc(300px + 12px);
  min-height: calc(60px + 12px);
  box-shadow: 0 0 60px rgba(0, 255, 203, 0.64);
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  opacity: 0;
  transition: all 0.3s ease-in-out 0s;
}

.button:hover,
.button:focus {
  color: #313133;
  transform: translateY(-6px);
}

.button:hover::before,
.button:focus::before {
  opacity: 1;
}

.button::after {
  content: '';
  width: 30px;
  height: 30px;
  border-radius: 100%;
  border: 6px solid #00ffcb;
  position: absolute;
  z-index: -1;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  animation: ring 1.5s infinite;
}

.button:hover::after,
.button:focus::after {
  animation: none;
  display: none;
}

@keyframes ring {
  0% {
    width: 30px;
    height: 30px;
    opacity: 1;
  }
  100% {
    width: 300px;
    height: 300px;
    opacity: 0;
  }
}
