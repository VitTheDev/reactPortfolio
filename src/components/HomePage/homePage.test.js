import React from 'react';
import { shallow } from 'enzyme';
import { Link, MemoryRouter } from 'react-router-dom';
import HomePage from './HomePage';

describe('HomePage Component', () => {
  it('renders a div with an h2 inside', () => {
    const wrapper = shallow(<HomePage />);
    // console.log(wrapper.debug());
    expect(wrapper.find('div h2')).toHaveLength(1);
  });
  it('renders', () => {
    const wrap = shallow(<HomePage />);
    expect(wrap).toMatchSnapshot();
  });
  it('renders a link to MyWork Component', () => {
    const wrapper = shallow(<HomePage />);
    expect(wrapper.find(Link)).toHaveLength(1);
  });
});
