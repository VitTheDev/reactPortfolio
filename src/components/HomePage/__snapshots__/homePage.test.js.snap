// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component renders 1`] = `ShallowWrapper {}`;

exports[`HomePage Component works 1`] = `ShallowWrapper {}`;
